server-spi-private/src/main/java/org/keycloak/authorization/AdminPermissionsSchema.java

```diff
@@ -388,4 +391,96 @@ public void addUResourceTypeResource(KeycloakSession session, ResourceServer res
             }
         }
     }
+
+    public void removeResourceObject(AuthorizationProvider authorization, ProviderEvent event) {
+        if (!isAdminPermissionsEnabled(authorization.getRealm()) || authorization.getRealm().getAdminPermissionsClient() == null) return;
+
+        String type;
+        String configKey;
+        String id;
+        if (event instanceof UserRemovedEvent userRemovedEvent) {
+            type = "user";
+            configKey = "users";
+            id = userRemovedEvent.getUser().getId();
+        } else if (event instanceof ClientRemovedEvent clientRemovedEvent) {
+            type = "client";
+            configKey = "clients";
+            id = clientRemovedEvent.getClient().getId();
+        } else if (event instanceof GroupRemovedEvent groupRemovedEvent) {
+            type = "group";
+            configKey = "groups";
+            id = groupRemovedEvent.getGroup().getId();
+        } else if (event instanceof RoleRemovedEvent roleRemovedEvent) {
+            type = "role";
+            configKey = "roles";
+            id = roleRemovedEvent.getRole().getId();
+        } else {
+            return;
+        }
+
+        ResourceServer server = authorization.getStoreFactory().getResourceServerStore().findByClient(authorization.getRealm().getAdminPermissionsClient());
+
+        Map<Policy.FilterOption, String[]> attributes = Map.of(
+                Policy.FilterOption.TYPE, new String[] {type},
+                Policy.FilterOption.CONFIG, new String[] {configKey, id}
+        );
+
+        List<Policy> policies = authorization.getStoreFactory().getPolicyStore().find(server, attributes, null, null);
```

Comment:
IIRC, we don't need this logic to remove realm resources from the policy config. The policies will be eventually updated whenever they are updated and the invalid entries are ignored when running evaluations or building representations.

Reply:
Wasn't aware of this, thanks. Changed it so that we now remove the policy only if the removed object is the last in the config.

Reply:
Turned out there is issue in UI. Created https://github.com/keycloak/keycloak/issues/38020 and removed the policies handling from this PR